### DISCUSSION TITLE
Invisible barrier blocking avatars from passing

### DISCUSSION THREAD

**Discussion by hilarytsai** [mozilla/hubs#2769] [General Questions & Discussion] [answered]
I walked around my scene, and there's this big area which for some reason, won't allow me to walk through.  The floor is walkable and my avatar can walk anywhere else; but it's like something invisible is blocking the way. In case it's relevant, the "floor" is actually a png that is laid just on top of the floor plain. But I don't think that's causing the issue.  Is this to do with having a big scene file, perhaps?

**Comment by rawnsley** [comment, marked as answer]
For background, [this page](https://hubs.mozilla.com/docs/physics-and-navigation.html) describes how the physics works in Hubs. Basically there are two elements: a nav-mesh that the player's feet stick to like fly paper and a collision mesh that they bounce off.

If you make the Floor Plan node visible in Spoke you will be able to see both of these. The nav-mesh will be solid blue and the collision mesh will either be red wireframe (if it's a detailed tri-mesh) or yellow wireframe (if it's an approximated height field).

If you enter _/debug_ into the chat field in a live room you will also see physics meshes and may glean some extra information.

Basically you are looking for a hole in the nav-mesh or an unexpected lump in the collision mesh at the place where your players are being denied entry.

**Reply by hilarytsai** [reply]
Thank you rawnsley!  It's been massively helpful reading about the floor plan and how to debug the nav-mesh.  Also, I'm stupid because I realized the issue was an encapsulating building marked invisible, keeping me from passing beyond..
